src/Npgsql/NpgsqlDatabaseInfo.cs

```diff
@@ -220,7 +220,9 @@ public static void RegisterFactory(INpgsqlDatabaseInfoFactory factory)
         {
             if (factory == null)
                 throw new ArgumentNullException(nameof(factory));
+
             Factories.Insert(0, factory);
```

Comment:
It's not a problem with this pr, but if we attempt to register a factory while concurrently loading `DatabaseInfo`, there will be an error.

Reply:
Am aware that `Factories` isn't thread safe (:

Since there's an issue with our tests only we can fix it later.

Reply:
We definitely don't *have* to support thread safety on this operation, but taking a lock here would be trivial so why not.